ui/tests/integration/components/project-input-variables-list-test.ts

```diff
@@ -0,0 +1,61 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, fillIn, click, pauseTest } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import { create, collection, clickable, isPresent, fillable } from 'ember-cli-page-object';
+
+const page = create({
+  hasForm: isPresent('[data-test-input-variables-form]'),
+  variablesList: collection('[data-test-input-variables-list-item]', {
+    dropdown: clickable('[data-test-input-variables-dropdown]'),
+    dropdownDelete: clickable('[data-test-input-variables-dropdown-delete]'),
+  }),
+  createButton: clickable('[data-test-input-variables-add-variable]'),
+  cancelButton: clickable('[data-test-input-variables-edit-cancel]'),
+  saveButton: clickable('[data-test-input-variables-edit-save]'),
+  varName: fillable('[data-test-input-variables-var-name]'),
+  varStr: fillable('[data-test-input-variables-var-str]'),
+});
+
+module('Integration | Component | project-input-variables-list', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders', async function (assert) {
```

Comment:
Is there a reason to keep all this on one test? It might be clearer to have multiple tests for the different asserts at the bottom, so we could see clearly if a test fails what specific logic has broken. Also helpful in case someone needs to write new tests, they can easily read the test name to tell if the logic they are trying to test is already covered or not

Reply:
Sabrina makes some really good points here. We should definitely clarify the name of this test, at least.

I’d say in tests for this sort of component (with significant behavior and state) we want two kinds of tests:

1. **Pure-rendering** — given these inputs, I except this DOM
2. **Flows** — given these inputs and this sequence of events, I expect this DOM and these side-effects

It’s also useful to write what I’d call “**integrity**” tests — given these inputs and this sequence of events, I expect state not to have leaked or fallen out of sync. With these, we have to figure out ways to *observe* if internal state has became out-of-sync or leaked, as we did on the regression tests for `<ProjectRepositorySettings>`.

Reply:
Yeah,  this is more of a user-flow test. I used the default test title and never changed it. 🙈 Maybe I'll add a simple rendering test for the empty state and an example initial variable set, to separate the concerns.

Reply:
Added a base rendering test separated from the user flow